src/ecs/resources/input/axis.rs

```diff
@@ -0,0 +1,11 @@
+use super::Button;
+
+/// Represents an axis made up of digital inputs, like W and S or A and D.
+/// Two of these could be analogous to a DPAD.
+#[derive(Serialize, Deserialize)]
+pub struct Axis {
+    /// Positive button, when pressed down axis value will return 1 if `neg` is not pressed down.
```

Comment:
I like the idea, we already have pretty good support for bool and float constraint, although the float constraint could be improved upon to be more generic rather than using strictly boolean inputs.  `float unconstrained` seems like a good idea for mouse stuff but I think I already emulate that pretty well with `mouse_position` and `mouse_position_change`.  What other input devices have you encountered that needed `float unconstrained`?